src/components/List.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import Card from "./Card";
+
+export default function List({ data, buttonText, onStateChange }) {
+  // function to move items between lists
+  const handleDeviceListClick = (id) => {
+    // get the index of the item that was clicked and pass it to our state handler
+    // match by uid
+    let itemToMoveIndex = data.map((el) => el.uid).indexOf(id);
+
+    console.log("handle click");
+    console.log(id);
+    console.log(data);
+    console.log("itemToMoveIndex = " + itemToMoveIndex);
+
+    onStateChange(itemToMoveIndex);
```

Comment:
Minor suggestion, but onStateChange() is kind of a generic name for this prop, a more detailed name like `itemClicked` might be better.